dreadnode/agent/agent.py

```diff
@@ -488,10 +490,16 @@ async def _dispatch(event: AgentEvent) -> t.AsyncIterator[AgentEvent]:  # noqa:
             raise winning_reaction
 
         # Tool calling
+        tool_calls = 0
 
         async def _process_tool_call(
             tool_call: "rg.tools.ToolCall",
         ) -> t.AsyncGenerator[AgentEvent, None]:
+            nonlocal tool_calls
+
+            if self.max_tool_calls != -1 and tool_calls >= self.max_tool_calls:
+                raise Finish("Reached maximum allowed tool calls.")
```

Comment:
Curious to understand, 

Do we need the duplicate `max_tool_calls` check?

i think the early check makes bottom check unreachable dead code. Consider removing this to be consistent with steps? WDYT?